### PR TITLE
scaling down to zero restrictions for RKE1

### LINKED ISSUE #5774: [rke1] user is able to scale dedicated `etcd` machine pool down to `0`  using the `...` option
**Setup**
- Rancher version: v2.6.5-rc5
- Browser type & version: chrome

**Describe the bug**
UI button is greyed out, however the `...` menu for the node allows a user to scale down any node to `0`

**To Reproduce**
* deploy an rke1 cluster, with dedicated nodes-per-role (i.e. 1 pool per role)
* once the cluster is active, select the cluster from the `cluster management` page and click the `...` option next to a node.
* click the option to scale down on the etcd node

**Result**
etcd node scales down, and cluster is down

**Expected Result**
user should not be able to scale etcd, controlplane to 0

**Screenshots**
Screen Shot 2022-04-28 at 6 20 19 PM

**Additional context**
this is not an option on rke2 clusters.

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Fixes #5774 

### Technical notes summary

Followed the same rules applied to scaling down RKE2 machines/nodes by reusing the same helper function `notOnlyOfRole` in order to disable/enable scaling down options. Decided to create a new prompt modal component `ScaleRke1NodeDownDialog.vue` to avoid adding more complexity to the current `ScaleMachineDownDialog.vue`.

### Areas or cases that should be tested

" - Scale Down" button on top of the table.
Minus buttons on the right side of each pool.
Scale down option from the dropdown menu when clicking on the three dots option on the right side of each node row.

### Screenshot/Video

https://github.com/rancher/dashboard/assets/135728925/97aed2c6-9233-4837-8419-fc0266456663

https://github.com/rancher/dashboard/assets/135728925/5a1cdbc7-78c7-4fa8-a317-a0cb465610ec

### CODE REVIEW

**Review by codyrancher — APPROVED** [review]

**Comment by richard-cox** [comment]
When adding the rke2 scale down feature there were a few tricky bits. @momesgin i'd like to check those out in rke1 land

**Comment by momesgin** [comment]
> When adding the rke2 scale down feature there were a few tricky bits. @momesgin i'd like to check those out in rke1 land

@richard-cox sounds good, let me know if you need anything from me

**Review by richard-cox — CHANGES_REQUESTED** [review]
Gave this a functional test, didn't cover every edge case but looks looks generally.

File: shell/dialog/ScaleRke1NodeDownDialog.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
+<script>
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/dialog/ScaleRke1NodeDownDialog.vue:1]: Other than the three cases that you mentioned above there's one more case when pointing to `cluster`:

```
const cluster = this.isRke2 ? this.resources[0].cluster : this.resources[0].provisioningCluster;
```

and `name` is actually `undefined` in the RKE1 `resources` object that we pass as a prop to the modal component.

My intention was to avoid adding more logic to the component specially since RKE1 could get deprecated in the future, but overall I don't have a strong opinion here so I combined them.

File: shell/dialog/ScaleMachineDownDialog.vue [hunk 2]
```diff
@@ -13,31 +13,41 @@ export default {
...
+        this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE_POOL }),
+        this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE })
+      ]);
+    }
   },
 
   data() {
-    const allToDelete = Array.isArray(this.resources) ? this.resources : [this.resources];
-    const cluster = allToDelete[0].cluster;
+    const cluster = this.isRke2 ? this.resources[0].cluster : this.resources[0].provisioningCluster;
```
Comment (momesgin) [inline shell/dialog/ScaleMachineDownDialog.vue:35]: @richard-cox the safety check for the resources to be an array is removed because it was a duplicate, we already have it on `PromptModal.vue`
https://github.com/rancher/dashboard/blob/91d756af80975b448f0a3d3424392c9686ee0362/shell/components/PromptModal.vue#L22

**Review by richard-cox — CHANGES_REQUESTED** [review]
One inline comment, plus i get an error when trying to scale down an individual rke2 machine. This is due to the `created` hook running after `data` which depends on it to find `cluster` .. which then goes on to error as it's not found. Moving that into data should work

File: shell/dialog/ScaleMachineDownDialog.vue [hunk 3]
```diff
@@ -103,7 +122,7 @@ export default {
     <template slot="body">
       <div class="pl-10 pr-10 mt-20 mb-20 body">
         <div v-if="allToDelete.length === 1">
-          {{ t('promptRemove.attemptingToRemove', { type }) }} <b>{{ safeMachinesToDelete[0].name }}</b>
+          {{ t('promptRemove.attemptingToRemove', { type }) }} <b>{{ isRke2 ? safeMachinesToDelete[0].name : safeMachinesToDelete[0].nameDisplay }}</b>
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/dialog/ScaleMachineDownDialog.vue]: I wasn't sure how to reproduce the error you've mentioned above, but I moved the logic from `created` block to `data` block. `nameDisplay` is also used for both rke1 and rke2 now, thank you for pointing these out 🙏

**Review by richard-cox — APPROVED** [review]
